module/extension/Director/src/Director.cpp

```diff
@@ -174,9 +175,9 @@ namespace module::extension {
         on<Trigger<DirectorTask>, Sync<DirectorTask>>().then(
             "Director Task",
             [this](std::shared_ptr<const DirectorTask> task) {
-                // Root level task, make the TaskPack immediately and send it off to be executed as a root task
+                // Root level task, make the pack immediately and send it off to be executed as a root task
                 if (providers.count(task->requester_id) == 0) {
-                    emit(std::make_unique<TaskPack>(TaskPack({task})));
+                    emit(std::make_unique<TaskList>(TaskList({task})));
```

Comment:
Do you really need to construct it? Doesnt make_unique call the constructor for you?

Reply:
It wouldn't compile when I did that. I did try

Reply:
c++ cannot work out the type of `{task}` is meant to be. We can either tell it or avoid using the initializer_list.
```suggestion
                    emit(std::make_unique<TaskList>(std::initializer_list({task})));
```
```suggestion
                    emit(std::make_unique<TaskList>(1, task));
```
The way you have it currently is the most readable I recon.